modules/ROOT/pages/con_pipeline-pull-request-resources.adoc

```diff
@@ -0,0 +1,125 @@
+[]
+= Pull Request Resource
+
+Pull Request resource represents a pull request event from a source control
+system.
+
+Adding the Pull Request resource as an input to a Task will populate the
+workspace with a set of files containing generic pull request related metadata
+such as base/head commit, comments, and labels.
+
+The payloads will also contain
+links to raw service-specific payloads where appropriate.
+
+Adding the Pull Request resource as an output of a Task will update the source
+control system with any changes made to the pull request resource during the
+pipeline.
+
+Example file structure:
+
+[source]
+----
+/workspace/
+/workspace/<resource>/
+/workspace/<resource>/labels/
+/workspace/<resource>/labels/<label>
+/workspace/<resource>/status/
+/workspace/<resource>/status/<status>
+/workspace/<resource>/comments/
+/workspace/<resource>/comments/<comment>
+/workspace/<resource>/head
+/workspace/<resource>/base
+----
+
+More details:
+
+Labels are empty files, named after the desired label string.
+
+Status is represented as a set of json files, structured like this:
+
+```json
+{
+  "ID": "foobar",
+  "Description": "description",
+  "URL": "https://www.example.com",
+  "Code": "success" 
+}
+```
+
+References (head and base) are a set of json files, structured like this:
+
+```json
+{
+  "Repo": "https://gitlab.com/foo/bar",
+  "Branch": "master",
+  "SHA": "b813c8fcb1e6245dcbe5ab3c14259fac2e75a799"
+}
+```
+
+Comments are a set of json files, structured like this:
+```json
+{
+  "Text": "comment body",
+  "Author": "author",
+  "ID": 202131633
+}
+```
+
+See [types.go](/cmd/pullrequest-init/types.go) for the full payload spec.
+
+To create a pull request resource using the `PipelineResource` CRD:
+
+```yaml
+apiVersion: tekton.dev/v1alpha1
+kind: PipelineResource
+metadata:
+  name: wizzbang-pr
+  namespace: default
+spec:
+  type: pullRequest
+  params:
+    - name: url
+      value: https://github.com/wizzbangcorp/wizzbang/pulls/1
+  secrets:
+    - fieldName: githubToken
+      secretName: github-secrets
+      secretKey: token
+```
+
+Params that can be added are the following:
+
+* `url`: represents the location of the pull request to fetch.
+
+== Statuses
+
+The following status codes are available to use for the Pull Request resource:
+
+// Will be a table
+// Status          
+// success         
+// neutral         
+// queued          
+// in_progress     
+// failure         
+// unknown         
+// error           
+// timeout         
+// canceled       
+// action_required
+
+
+[NOTE]
+====
+Status codes are currently **case sensitive**.
+====
```

Comment:
@joaedwar: Is this the convention you follow for Note? In Asciidoc, we usually simply say: 
NOTE: Status codes are currently **case sensitive**.